src/autoQuery/units.ts

```diff
@@ -16,20 +28,74 @@ export function getUnitFromMetric(metric: string) {
     return null;
   }
 
-  const metricParts = metric.toLowerCase().split('_').slice(-2); // Get last two parts
-  for (let i = metricParts.length - 1; i >= 0; i--) {
-    if (UNIT_LIST.includes(metricParts[i])) {
-      return metricParts[i];
+  const metricLower = metric.toLowerCase();
+
+  // First check for special patterns like CPU metrics
+  if (metricLower.includes('cpu_seconds') || metricLower.includes('cpu_seconds_total')) {
+    return 'seconds';
+  }
+
+  // Check for duration or latency metrics
+  if (metricLower.includes('duration') || metricLower.includes('latency')) {
+    return 'seconds';
+  }
+
+  // Then try the suffix-based approach
+  const metricParts = metricLower.split('_');
+
+  // Check the last two parts first (common pattern)
+  for (let i = metricParts.length - 1; i >= Math.max(0, metricParts.length - 2); i--) {
+    const part = metricParts[i];
+    if (UNIT_LIST.includes(part)) {
+      return part;
+    }
+  }
+
+  // Check for common units anywhere in the metric name
+  for (const unit of UNIT_LIST) {
+    if (metricLower.includes(unit)) {
+      return unit;
     }
   }
+
+  // Check for common patterns
+  if (metricLower.endsWith('_ms') || metricLower.includes('_ms_')) {
+    return 'milliseconds';
+  } else if (metricLower.endsWith('_count') || metricLower.includes('_count_')) {
+    return 'count';
+  } else if (metricLower.includes('percentage') || metricLower.includes('ratio')) {
+    return 'percent';
```

Comment:
same here, i just get nervous when returning strings like that since it can be a bit more error prone in the future for tests or refactors